### DISCUSSION TITLE
Is it possible to render a <MenuItem /> as a <DialogDisclosure />?

### DISCUSSION THREAD

**Discussion by crisshaker** [ariakit/ariakit#3943] [Questions] [answered]
I'm looking for a declarative way to make a MenuItem trigger a Dialog on click.
I already understand this is possible to do programmatically.
I tried using the render API but couldn't get that to work. Is this behavior officially supported?

**Comment by diegohaz** [comment, marked as answer]
Yes:
```jsx
<Ariakit.DialogProvider>
  <Ariakit.MenuItem render={<Ariakit.DialogDisclosure />}>Open</Ariakit.MenuItem>
  <Ariakit.Dialog>Content</Ariakit.Dialog>
</Ariakit.DialogProvider>
```

**Reply by crisshaker** [reply]
Ohh great. It works!!! However, I had to explicitly pass the dialogStore into both DialogProvider and DialogDisclosure (see code snippet) even though I had a DialogProvider; otherwise, I get the error in the attached image. 

```tsx
function TestButton() {
  const dialogStore = useDialogStore();
  const menuStore = useMenuStore();

  return (
    <DialogProvider store={dialogStore}>
      <MenuButton store={menuStore}>Menu</MenuButton>
      <Menu store={menuStore}>
        <MenuItem render={<DialogDisclosure store={dialogStore} />}>Open Dialog</MenuItem>
      </Menu>

      <Dialog
        backdrop={<div className="fixed inset-0 z-40 bg-black/60" />}
        className="fixed inset-32 z-40 rounded-lg bg-white"
      />
    </DialogProvider>
  );
}
```
<img width="1156" alt="image" src="https://github.com/user-attachments/assets/0d51f191-9dbc-40a4-9437-9896bc0da527">

**Reply by diegohaz** [reply]
I see. In that case, I suggest you use `useState` instead of `useDialogStore`, just like the main [Dialog example](https://ariakit.org/components/dialog).

**Reply by crisshaker** [reply]
Got it! Thanks!
